### PR TITLE
[INF] Install pre-commit into repository #601

### LINKED ISSUE #601: [INF] Add pre-commit hooks to repository
Pre-commit hooks are a great way to push code style checks to the development machine, so that they automatically happen and thus we don't have to wait for the CI system to finish to  discover that it failed only on code style.

I will add in instructions to get started with `pre-commit`,  so that a bunch of checks are automatically done pre-commit.

### PR DESCRIPTION
**Pull request by ericmjl**
# PR Description

Please describe the changes proposed in the pull request: 

- A bunch of code formatting happened. Do not fret over the thousands of line changes; most of  them are just cosmetic changes.
- The substantial changes are in the `Makefile` and in `.pre-commit-config.yaml`.

**This PR resolves #601.**

Reviews requested from @zbarry @hectormz and @szuckerman, just want to make sure at least one of you know what changes have happened.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -36,7 +35,17 @@ This also installs your new conda environment as a Jupyter-accessible kernel. To
...
 .. note:: For PyCharm users, here are some `instructions <PYCHARM_USERS.html>`__  to get your Conda environment set up.
 
-4. You should also be able to build the docs locally. To do this, from the main ``pyjanitor`` directory::
+4. (Optional) Install the pre-commit hooks.
+
+As of 29 October 2019, pre-commit hooks are available to run code formatting checks automagically
+before git commits happen. If you did not have these installed before, run the following commands:
+
+    # Update your environment to install pre-commit
+    $ conda env update  -f environment.yml
```
Comment (hectormz) [inline CONTRIBUTING.rst]: Shouldn't this be `environment-dev.yml`?

**Comment by ericmjl** [comment]
@hectormz you can do the honors :smile:

**Comment by hectormz** [comment]
Gladly!

**Comment by zbarry** [comment]
I'm a fan. Thanks for putting this in.

**Comment by zbarry** [comment]
@ericmjl - uhhhhhh, is this a desirable thing?

image

**Comment by hectormz** [comment]
@zbarry @ericmjl looks like it removed trailing tabs. `test_join_fasta` which uses that file was successful, and the file opens fine locally for me, so I suppose it is okay? But not as fruitful as formatting code?

**Pull request merged**